homeassistant/components/influxdb.py

```diff
@@ -89,56 +92,54 @@ def setup(hass, config):
     def influx_event_listener(event):
         """Listen for new messages on the bus and sends them to Influx."""
         state = event.data.get('new_state')
-        if state is None or state.entity_id in blacklist:
-            return
-
-        if whitelist and state.entity_id not in whitelist:
+        if state is None or state.state in (
+                STATE_UNKNOWN, '', STATE_UNAVAILABLE) or \
+                state.entity_id in blacklist:
             return
 
         try:
+            if len(whitelist) > 0 and state.entity_id not in whitelist:
+                return
+
             _state = state_helper.state_as_number(state)
+            _state_key = "value"
         except ValueError:
             _state = state.state
+            _state_key = "state"
+
+        measurement = state.attributes.get('unit_of_measurement')
+        if measurement in (None, ''):
+            if default_measurement:
```

Comment:
Not sure if this was the original or not. But this means you can only override the default if a unit_of_measurement is not set.